addons/website_event_track/models/event_track.py

```diff
@@ -630,3 +648,91 @@ def _get_track_suggestions(self, restrict_domain=None, limit=None):
+                'therefore the times indicated in this reminder correspond to those of the event.'
+            )
+        } if not self.date else ''
+
+    def _get_external_description(self):
+        """ Adding the URL of the event track into the description """
+        self.ensure_one()
+        return Markup("<a href='%(event_track_url)s'>%(name)s</a>\n%(shorten_description)s\n\n%(reminder_times_warning)s") % {
+            'event_track_url': werkzeug.urls.url_join(self.get_base_url(), self.website_url),
+            'name': self.name,
+            'shorten_description': shorten(html_to_inner_content(self.description), 1900),
+            'reminder_times_warning': self._get_reminder_times_warning(),
+        }
+
+
+    def _get_event_track_reminder_dates(self):
+        return {
+            'date_begin': self.date or self.event_id.date_begin,
+            'date_end': self.date_end or self.event_id.date_end,
+        }
+
+    def _get_yahoo_resource_url(self):
+        date_tz = self.event_id.date_tz
+        reminder_dates = self._get_event_track_reminder_dates()
+
+        params = {
+            'title':  f'{self.event_id.name}: {self.name}',
+            'in_loc': self.location_id.name,
+            'st': reminder_dates['date_begin'].astimezone(timezone(date_tz)).strftime('%Y%m%dT%H%M%S'),
+            'et': reminder_dates['date_end'].astimezone(timezone(date_tz)).strftime('%Y%m%dT%H%M%S'),
+            'desc': shorten(html_to_inner_content(self.description), 1900) + f'\n\n{html_to_inner_content(self._get_reminder_times_warning())}'
+        }
+        return YAHOO_CALENDAR_URL + werkzeug.urls.url_encode(params)
+
+    def _get_event_track_resource_urls(self):
+        date_tz = self.event_id.date_tz
+        reminder_dates = self._get_event_track_reminder_dates()
+        url_date_begin = reminder_dates['date_begin'].astimezone(timezone(date_tz)).strftime('%Y%m%dT%H%M%S')
+        url_date_end = reminder_dates['date_end'].astimezone(timezone(date_tz)).strftime('%Y%m%dT%H%M%S')
+
+        params = {
+            'action': 'TEMPLATE',
+            'text': f'{self.event_id.name}: {self.name}',
+            'dates': f'{url_date_begin}/{url_date_end}',
+            'ctz': self.event_id.date_tz,
+            'details': self._get_external_description(),
+        }
+
+        if self.location_id:
+            params.update(location=self.location_id.sudo().name)
+
+        encoded_params = werkzeug.urls.url_encode(params)
+        google_url = GOOGLE_CALENDAR_URL + encoded_params
+        iCal_url = f'{self.get_base_url()}/event/{self.event_id.id}/track/{self.id}/ics?{encoded_params}'
+        return {'google_url': google_url, 'iCal_url': iCal_url, 'yahoo_url': self._get_yahoo_resource_url()}
+
+    def _get_ics_file(self):
+        """ Returns iCalendar file for the event track.
+            :returns a dict of .ics file content for each event track
+        """
+        result = {}
+        if not vobject:
+            return result
```

Comment:
Note that returning the same type as result (aka one key / record) allows to always have a value for each record id, like
```suggestion
            return dict.fromkeys(self, False)
```